gui/wxpython/lmgr/frame.py

```diff
@@ -1172,6 +1166,27 @@ def ChangeMapset(self, mapset):
                 display.SetTitleWithName(str(dispId))  # TODO: signal ?
                 dispId += 1
 
+    def SwitchMapset(self, grassdb, location, mapset):
+        """Switch to location and mapset"""
+        gisenv = grass.gisenv()
+        if can_switch_mapset_interactive(self, grassdb, location, mapset):
+            # Switch to mapset in the same location
+            if (grassdb == gisenv['GISDBASE'] and location == gisenv['LOCATION_NAME']):
+                self.datacatalog.changeMapset.emit(mapset=mapset)
+            # Switch to mapset in the same grassdb
+            elif grassdb == gisenv['GISDBASE']:
+                self.datacatalog.changeLocation.emit(mapset=mapset,
+                                                     location=location,
+                                                     dbase=None)
```

Comment:
This is an obscure way of calling self.ChangeLocation() - datacatalog emits a signal and it's connected in lmgr/frame.py which calls ChangeLocation().